File: Backend/TravelPlanner.Core/DomainModels/WeatherForecast.cs
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace TravelPlanner.DomainModels
{
    [DataContract]
    public class WeatherForecast
    {
        [DataMember]
        public DateTime Date { get; set; }

        [DataMember]
        public double WindSpeed { get; set; }

        [DataMember]
        public double WindDirection { get; set; }

        [DataMember]
        public double Visibility { get; set; }

        [DataMember]
        public double Cloudiness { get; set; }

        [DataMember]
        public IEnumerable<WeatherProperties> Weather { get; set; }

        [DataMember]
        public double Temperature { get; set; }

        [DataMember]
        public double TemperatureFeels{ get; set; }

        [DataMember]
        public double MinimalTemperature { get; set; }

        [DataMember]
        public double MaximalTemperature { get; set; }

        [DataMember]
        public double Pressure { get; set; }

        [DataMember]
        public double Humidity { get; set; }

        public class WeatherProperties
        {
            [DataMember]
            public string WeatherName { get; set; }

            [DataMember]
            public string WeatherDescription { get; set; }

            [DataMember]
            public string WeatherIcon { get; set; }
        }
    }
}

File: TravelPlannerApp/app/src/main/java/com/travelplanner/models/WeatherForecast.kt
package com.travelplanner.models

import java.time.LocalDateTime

class WeatherForecast(val date: LocalDateTime, val windSpeed: Double, val windDirection: Double,
                      val visibility: Double, val cloudiness: Double, val weather: List<WeatherProperties>,
                      val temperature: Double, val temperatureFeels: Double, val minimalTemperature: Double,
                      val maximalTemperature: Double, val pressure: Double, val humidity: Double)

class WeatherProperties(val weatherName: String, val weatherDescription:String, val weatherIcon: String)